app/src/main/java/at/aau/serg/activities/JoinLobbyActivity.kt

```diff
@@ -0,0 +1,155 @@
+            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
+            insets
+        }
+
+        val recyclerView = findViewById<View>(R.id.recyclerViewLobbies) as RecyclerView
+        adapter = JoinLobbyLobbiesAdapter(this, lobbies)
+        recyclerView.setHasFixedSize(true)
+        recyclerView.setLayoutManager(LinearLayoutManager(this))
+        recyclerView.setAdapter(adapter)
+
+
+        HttpClient.get(
+            "lobbys",
+            StoreToken(this).getAccessToken(),
+            CallbackCreator().createCallback(::onFailureLobby, ::onSuccessGetLobbies)
+        )
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    private fun onSuccessGetLobbies(response: Response) {
+        if (response.isSuccessful) {
+            val responseBody = response.body?.string()
+            if (responseBody != null) {
+                val openLobbies = Array(JSONArray(responseBody).length()) {
+                    JSONArray(responseBody)[it]
+                }
+                for ((index, lobby) in openLobbies.withIndex()) {
+                    Log.d("LO", lobby.toString())
+                    val jsonLobby = JSONObject(lobby.toString())
+                    lobbies.add(
+                        index, JoinLobbyLobby(
+                            jsonLobby.getString("name"),
+                            jsonLobby.getJSONArray("players").length(),
+                            jsonLobby.getInt("maxPlayers"),
+                            jsonLobby.getString("lobbyid")
+                        )
+                    )
+                }
+            }
+            runOnUiThread {
+                adapter.notifyDataSetChanged()
+            }
+
+            Log.d("LO", lobbies.toString())
+        }
+    }
+
+
+    fun btnJoinCode(view: View) {
+        jLobbyWithCode(null)
+    }
+
+    private fun jLobbyWithCode(response: Response?) {
+        if (lobbyToJoin.lobbyID != "") {
+            joinLobby(lobbyToJoin)
+            return
+        }
+        val builder: AlertDialog.Builder = AlertDialog.Builder(this)
+        builder.setTitle("Enter LobbyID")
+        val input = EditText(this)
+
+        input.setInputType(InputType.TYPE_CLASS_TEXT)
+        builder.setView(input)
+        builder.setPositiveButton(
+            "OK"
+        ) { _, _ ->
+            lobbyToJoin = LobbyJoin(input.getText().toString())
+            joinLobby(lobbyToJoin)
+        }
+        builder.setNegativeButton(
+            "Cancel"
+        ) { dialog, _ -> dialog.cancel() }
+        builder.show()
+    }
+
+    private fun joinLobby(lobbyToJoin: LobbyJoin) {
+        HttpClient.post(
+            "/lobbys/join",
+            Gson().toJson(lobbyToJoin),
+            StoreToken(this).getAccessToken(),
+            CallbackCreator().createCallback(::onFailureLobby, ::onSuccessJoinLobby)
+        )
+    }
+
+    private fun onFailureLobby() {
```

Comment:
We should probably add some more specific error handling, but we could also do that in a separate PR where we change the ` createCallback`  to include the error.

Reply:
Yes i would move that to the other pr